### PR TITLE
restricted shell: check that LSM BPF is enabled

### PR DESCRIPTION
**Pull request by eyakubovich**
Checks that /sys/kernel/security/lsm contains "bpf"
before starting restricted shell. Aborts with error
if the entry is not found.

### CODE REVIEW

File: lib/restrictedsession/restricted.go [hunk 1]
```diff
@@ -298,6 +306,27 @@ func (l *auditEventLoop) close() {
 	l.wg.Wait()
 }
 
+// checkLsmBpf checks that eBPF is one of the enable
+// LSM "modules".
+func checkBpfLsm() error {
+	lsmInfo := "/sys/kernel/security/lsm"
+
+	csv, err := ioutil.ReadFile(lsmInfo)
```
Comment (jakule) [inline lib/restrictedsession/restricted.go:314]: nit: `ioutils` is deprecated:
```suggestion
	csv, err := os.ReadFile(lsmInfo)
```

File: lib/restrictedsession/restricted.go [hunk 2]
```diff
@@ -298,6 +306,27 @@ func (l *auditEventLoop) close() {
 	l.wg.Wait()
 }
 
+// checkLsmBpf checks that eBPF is one of the enable
+// LSM "modules".
+func checkBpfLsm() error {
+	lsmInfo := "/sys/kernel/security/lsm"
```
Comment (jakule) [inline lib/restrictedsession/restricted.go:312]: nit: can we change to constant?

File: lib/restrictedsession/restricted.go [hunk 3]
```diff
@@ -298,6 +306,27 @@ func (l *auditEventLoop) close() {
...
+		return trace.Wrap(err)
+	}
+
+	for _, mod := range strings.Split(string(csv), ",") {
+		if mod == "bpf" {
+			return nil
+		}
+	}
+
+	return fmt.Errorf(`%s does not contain bpf entry, indicating that the kernel
```
Comment (jakule) [inline lib/restrictedsession/restricted.go:325]: nit:
```suggestion
	return trace.Errorf(`%s does not contain bpf entry, indicating that the kernel
```

**Review by jakule — APPROVED** [review]
LGTM a few minor comments

File: lib/restrictedsession/restricted.go (hunk 2 above)
Comment (eyakubovich) [reply lib/restrictedsession/restricted.go:312]: Yep. For some reason I thought that Go doesn't support `const` at local level :facepalm:

**Comment by russjones** [comment]
@nklaassen Do you mind doing a review as well?

File: lib/restrictedsession/restricted.go [hunk 4]
```diff
@@ -298,6 +305,27 @@ func (l *auditEventLoop) close() {
 	l.wg.Wait()
 }
 
+// checkLsmBpf checks that eBPF is one of the enable
+// LSM "modules".
```
Comment (nklaassen) [inline lib/restrictedsession/restricted.go:309]:
```suggestion
// checkBpfLsm checks that eBPF is one of the enabled
// LSM "modules".
```

**Review by nklaassen — APPROVED** [review]
LGTM besides the comment nit

**Review by russjones — APPROVED** [review]

**Comment by r0mant** [comment]
@russjones Is this okay to merge? I'll enabled auto-merge if so.

File: lib/restrictedsession/restricted.go [hunk 5]
```diff
@@ -298,6 +305,27 @@ func (l *auditEventLoop) close() {
...
+	csv, err := os.ReadFile(lsmInfo)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, mod := range strings.Split(string(csv), ",") {
+		if mod == "bpf" {
+			return nil
+		}
+	}
```
Comment (zmb3) [inline lib/restrictedsession/restricted.go:322]: nit: we can do this without extra allocations

```suggestion
	for _, mod := range bytes.Split(csv, []byte(",")) {
		if bytes.Equal(mod, []byte("bpf")) {
			return nil
		}
	}
```

**Comment by russjones** [comment]
@r0mant If @nklaassen and @jakule signed off, let's merge.

**Review by r0mant — APPROVED** [review]

**Pull request merged**